### PR TITLE
Handle android app links

### PR DESCRIPTION
**Pull request by TonyGiorgio**

### CODE REVIEW

File: android/app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -18,11 +20,25 @@
             android:launchMode="singleTask"
             android:exported="true">
 
+            <tools:validation testUrl="https://android-app-links.mutiny-web.pages.dev/" />
```
Comment (TonyGiorgio) [inline android/app/src/main/AndroidManifest.xml:23]: TODO: remove this

File: android/app/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -18,11 +20,25 @@
...
 
+            <intent-filter>
+                <action android:name="android.intent.action.VIEW" />
+
+                <category android:name="android.intent.category.DEFAULT" />
+                <category android:name="android.intent.category.BROWSABLE" />
+
+                <data
+                    android:scheme="https"
+                    android:host="android-app-links.mutiny-web.pages.dev"
```
Comment (TonyGiorgio) [inline android/app/src/main/AndroidManifest.xml:38]: TODO: switch to app.mutinywallet.com

**Comment by TonyGiorgio** [comment]
This kind of works. I might need some help with two different problems: 

1. I don't think `App.tsx` lives forever. I get unmounts / cleanups whenever I move to the settings page, and if I minimize the app on the settings page and click on a url like `app.mutinywallet.com/settings/connections?callbackUri=primal&name=Primal` then it does not load the connections page. Is there a better place for these long lived global listeners to stay? 
2. The `Connections.tsx` page is not loading the query parameters if the app was already opened. I have verified that the `navigate` is working properly, but `name` is null on `Connection.txs`. If I force close the app and click on a link with query parameters, those make it across properly. 

Tried a bunch of random little things but couldn't get it working for these.

File: src/components/App.tsx [hunk 3]
```diff
@@ -30,25 +28,6 @@ export function App() {
     const i18n = useI18n();
     const [state, _actions] = useMegaStore();
 
-    // Check if the platform is Android to handle back
```
Comment (benthecarman) [inline src/components/App.tsx:33]: are you removing this on purpose?

Comment (benthecarman) [reply src/components/App.tsx:33]: oh nvm, i see you moved it

**Review by benthecarman — COMMENTED** [review]
the web links still work, otherwise can't review much else

**Review by benalleng — COMMENTED** [review]
tested on Pixel 4a using the debug.apk

with apps:

https://github.com/MutinyWallet/mutiny-web/assets/108441023/44ad87f0-c581-4012-aff6-6b3d2c2fdd8b

without apps:

https://github.com/MutinyWallet/mutiny-web/assets/108441023/89ae54f6-ee04-46a1-bd9b-bb249c070bf1

**Comment by TonyGiorgio** [comment]
> How should I test this after downloading the debug.apk?

click a link with https://app.mutinywallet.com and it should open the app. Click a link with https://app.mutinywallet.com/settings/connections?callbackUri=primal&name=Primal and it will open the connections page with a box to fill it in. 

However, it does require going into android settings > app > mutiny > default links and turning on https://app.mutinywallet.com because it doesn't have the deployed well known app links thing yet.

**Review by benalleng — COMMENTED** [review]
https://github.com/MutinyWallet/mutiny-web/assets/108441023/32833020-0259-4577-ad00-ab2998c8430e

Looks like almost all links work! small inconsistency in Primal

File: android/app/src/main/AndroidManifest.xml [hunk 4]
```diff
@@ -23,6 +25,18 @@
...
+            <intent-filter>
+                <action android:name="android.intent.action.VIEW" />
+
+                <category android:name="android.intent.category.DEFAULT" />
+                <category android:name="android.intent.category.BROWSABLE" />
+
+                <data
+                    android:scheme="https"
+                    android:host="app.mutinywallet.com"
+                    android:pathPattern="/.*" />
```
[the remark this answers is not in the document]
Comment (TonyGiorgio) [reply android/app/src/main/AndroidManifest.xml:37]: Interesting! It's just from the primal app that the URL doesn't open up the app when it doesn't have the forward slash at the end? It opens up from other apps?

[the remark this answers is not in the document]
Comment (benalleng) [reply android/app/src/main/AndroidManifest.xml:37]: on testing it looks like amethyst does the same thing, at least with those 2 links. It worked with no forward slash on github from the link you posted. gonna test them here.

https://app.mutinywallet.com
https://app.mutinywallet.com/

[the remark this answers is not in the document]
Comment (TonyGiorgio) [reply android/app/src/main/AndroidManifest.xml:37]: I added another intent link for one without a `/` at all to see if that helps.

[the remark this answers is not in the document]
Comment (benalleng) [reply android/app/src/main/AndroidManifest.xml:37]: hmm unfortunately, it doesn't seem to work. Though going through a bunch of different apps on my phone I am now convinced no-one really fully understands how to set them up properly, linkedin can't figure it out either lol

File: android/app/src/main/AndroidManifest.xml [hunk 5]
```diff
@@ -23,6 +25,21 @@
...
+                <data
+                    android:scheme="https"
+                    android:host="app.mutinywallet.com" />
+            </intent-filter>
```
Comment (benalleng) [inline android/app/src/main/AndroidManifest.xml:41]: Figured it out. I believe the data needs to be separate for it to work, additionally on my Android copying or opening any link like [app.mutinywallet.com](http://app.mutinywallet.com) adds `http://` not `https://` in front in the clipboard. I suggest we try something like this.

```suggestion
<intent-filter>
                <action android:name="android.intent.action.VIEW" />

                <category android:name="android.intent.category.DEFAULT" />
                <category android:name="android.intent.category.BROWSABLE" />

                <data
                    android:scheme="https"
                    android:host="app.mutinywallet.com" />
            </intent-filter>

            <intent-filter>
                <action android:name="android.intent.action.VIEW" />

                <category android:name="android.intent.category.DEFAULT" />
                <category android:name="android.intent.category.BROWSABLE" />

                <data
                    android:scheme="http"
                    android:host="app.mutinywallet.com" />
            </intent-filter>
```

**Review by benalleng — CHANGES_REQUESTED** [review]
I think I figured it out, tested on my test PR #493 and it works!

**Comment by TonyGiorgio** [comment]
Thanks @benalleng! Yeah that sounds right. I was hesitant to test the different path version one just out of laziness but I had seen that suggestion as well. Wasn't sure if it was going to make a different. But yeah I did read on spliting the intents before so good catch there.

**Review by benalleng — APPROVED** [review]
Still have one stray import warning but other than that it LGTM!

File: src/root.tsx [hunk 6]
```diff
@@ -10,15 +10,63 @@ import {
...
+    Title,
+    useNavigate
 } from "solid-start";
 
 // eslint-disable-next-line
 import "@mutinywallet/ui/style.css";
 
+import { App as CapacitorApp } from "@capacitor/app";
+import { Capacitor } from "@capacitor/core";
+import { onCleanup } from "solid-js";
```
Comment (futurepaul) [inline src/root.tsx:22]: need to merge this import with the one above... tbh odd that `pnpm run format` isn't doing this automatically

File: src/root.tsx [hunk 7]
```diff
@@ -10,15 +10,63 @@ import {
...
+            navigate(`${path}?${urlParams.toString()}`);
+        });
+
+        onCleanup(() => {
+            console.debug("cleaning up appUrlOpen listener");
+            remove();
+        });
+    }
+
+    return null;
```
Comment (futurepaul) [inline src/root.tsx:67]: this is so weird to me to make a null component but seems to work so not gonna fight it!

**Review by futurepaul — APPROVED** [review]
works great tested on device, and the `.well-known/assetlinks.json` seems like it deploys to the right place

**Pull request merged**